build.sbt

```diff
@@ -282,4 +284,125 @@ val generate = taskKey[Unit]("generate code from APIs")
 // Library versions
 val jacksonVersion = "2.17.0"
 val openApiToolsJacksonBindNullableVersion = "0.2.6"
-val log4jVersion = "2.23.1"
\ No newline at end of file
+val log4jVersion = "2.23.1"
+
+/*
+ ********************
+ * Release settings *
+ ********************
+ */
+import ReleaseTransformations._
+
+lazy val skipReleaseSettings = Seq(
+  publishArtifact := false,
+  publish / skip := true
+)
+
+// define getMajorMinorPatch function
+/**
+ * @return tuple of (major, minor, patch) versions extracted from a version string.
+ *         e.g. "1.2.3" would return (1, 2, 3)
+ */
+def getMajorMinorPatch(versionStr: String): (Int, Int, Int) = {
+  implicit def extractInt(str: String): Int = {
+    """\d+""".r.findFirstIn(str).map(java.lang.Integer.parseInt).getOrElse {
+      throw new Exception(s"Could not extract version number from $str in $version")
+    }
+  }
+
+  versionStr.split("\\.").toList match {
+    case majorStr :: minorStr :: patchStr :: _ =>
+      (majorStr, minorStr, patchStr)
+    case _ => throw new Exception(s"Could not parse version for $version.")
+  }
+}
+
+
+// Release settings for artifact that contains only Java source code
+lazy val javaOnlyReleaseSettings = releaseSettings ++ Seq(
```

Comment:
Lets add all of these as a scala file under projects/

Reply:
Done